### ISSUE TITLE
Try out new "swipe to focus" input

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#355]
We would like to try a new kind of input to support testing the self-voicing and vibration features with BVI participants. The input will behave similarly to VoiceOver control, and hopefully make the sim usable on touch screens when the screen cannot be seen. It should behave like
- Swipe right to focus next item.
- Swipe left to focus previous item.
- Double tap to 'click' item.
- Press and hold to begin dragging item.

**Comment by terracoda** [comment]
@jessegreenberg, I'd like to add to this issue, my notes:
- Double tap on interactive to activate or get instructions
  - buttons will activate
  - other interactions like sliders or custom interactions will provide instructions
- Then learner follows the instructions and we see if it works for them
- Possible Instructions for JT
  - Hold and rub finger to swing leg

If we want something more general, and we want to avoid the word "Press", we could try:
- Hold finger down and move to drag focused item.
- To {{move/rub/drag}} focused item, move finger without lifting.
- Hold down and move finger to move focused object.

Just brainstorming.

**Comment by jessegreenberg** [comment]
Thanks @terracoda, sounds good. I noticed that instead of "Press" VoiceOver uses "Tap" as the name for the gesture. FOr their drag interaction they say "Double-Tap and hold" in https://support.apple.com/guide/iphone/learn-voiceover-gestures-iph3e2e2281/ios. For sliders they have

> double-tap and hold the slider until you hear three rising tones, then drag the slider

**Comment by terracoda** [comment]
@jessegreenberg, I like the explicit indication (alert) that the object becomes "grabbed" and thus ready for interaction (dragging).

As discussed in the Self-Voicing meeting, it would be good to have an indication of the object being "Released", too, when the finger is lifted.

This is how I imagine it working:
- User swipes to go through focusable items, 
- On focus user hears name of object that is focused, e.g., "Leg Swing"
- User does an intuitive double tap to try to activate or to get more information. They hear the instructions for the focused item on double tap.
    - "Tap and hold to grab and drag."
- User executes tap and hold, and hears:
    - "Grabbed. Foot rubbing on rug." (Grabbed + {{current position}})
    - "Grabbed. Foot off rug." (When a grab happens and the foot is off the rug).
- User moves finger without lifting it, and hears:
     - rubbing sound + charge build-up sound
 - User lifts finger, and hears:
    - "Released." "Several electrons on body." "Leg Swing, foot rubbing on rug" 

Note that currently, a user needs to release the Leg Swing to get the new leg swing position and amount of electrons on body.

@jessegreenberg, how easy would it be to provide charge accumulation before a release?

**Comment by jessegreenberg** [comment]
That looks great @terracoda. The only different I would recommend is that focus with swipe should behave just like keyboard focus for self-voicing output. So I would change the second and fourth bullet points to 
- On focus user hears name of object that is focused and it's current value, e.g., "Leg Swing, Foot rubbing on rug."
- User executes tap and hold, and hears:
  - "Grabbed. Foot rubbing on rug."

Would that be OK?

> how easy would it be to provide charge accumulation before a release?

Very difficult at the moment unfortunately. I think it is time to work on either using utteranceQueue directly for self-voicing or create something like it for speech synthesis so we can control these things.

**Comment by terracoda** [comment]
@jessegreenberg, I like your suggestions for on focus and grabbed events.

As for graduating to utteranceQueue, I see now why that is important and also a very good idea.

**Comment by terracoda** [comment]
In https://github.com/phetsims/john-travoltage/issues/355#issue-716038715, you listed out:

>    Swipe right to focus next item.
>    Swipe left to focus previous item.
>    Double tap to 'click' item.
>    Press and hold to begin dragging item.

Do you think we will need to make instructions for swiping available, like we do for keyboard?
Or are these going to be simple enough we can get away with a general reference point/ reference document somewhere about the Self-Voicing feature?

Just jotting this adjustment down while I am thinking about it.
- Swipe right to focus next item.
- Swipe left to focus previous item.
- Double tap to activate item.
- Once activated, tap and hold to grab and drag.

I think "activate" is more general than "click".
For simple things like buttons and checkboxes, double tap will active or toggle, and for custom interactions like sliders, double tap will grab the item and provide further instructions.

**Comment by jessegreenberg** [comment]
> Do you think we will need to make instructions for swiping available, like we do for keyboard?

Good question, I don't know. I really hope that it is intuitive enough that it isn't necessary at all. For those that need the gestures, they should align almost completely with screen reader gestures that they are familiar with.

> I think "activate" is more general than "click".

Sounds good to me!

> Once activated, tap and hold to grab and drag.

There actually isn't an "activation" step required. The double tap currently just gives help information. The user can tap and hold any time to begin drag.

**Comment by jessegreenberg** [comment]
>  how easy would it be to provide charge accumulation before a release?

OK, now that we are using utterance-queue for self-voicing this has been done in #360. @terracoda is there anything else required for this issue?

**Comment by terracoda** [comment]
Swipe to focus works great for me in JT

I have 1 suggestion for improvement, and this can be for down the road.

I think it would be very useful to have an option to add the name of the thing that is to be grabbed and dragged, for example:
- Tap and hold to grab and drag {{hand}}.
- Tap and hold to grab and drag {{foot}}.

For Gravity Force Lab: Basics it might be much more needed than in JT:
- Tap and hold to grab and drag {{mass 1, blue sphere}}.
- Tap and hold to grab and drag {{mass 2, red sphere}}.

It GFLB this little detail provides an opportunity to bring in the word "sphere" and the "color" of each mass sphere which are prominent in the visual design and no present in the interactive part of the Self-Voicing. These details are available, of course in the read-me buttons.

**Comment by terracoda** [comment]
@jessegreenberg, shall we start a new issue for my last comment? There we can decide when and if we would make this improvement.

**Comment by terracoda** [comment]
Actually, I'll just make a new issue and close this one.
